pytroll_collectors/tests/test_scisys.py

```diff
@@ -216,14 +217,51 @@ def test_reception(self):
         os.remove(filename)
 
 
-def suite():
-    """Test suite for test_scisys."""
-    loader = unittest.TestLoader()
-    mysuite = unittest.TestSuite()
-    mysuite.addTest(loader.loadTestsFromTestCase(ScisysReceiverTest))
+def test_get_subject_from_msg2send_viirs():
+    """Test the get the subject from the message being send - viirs data."""
+    msg_rec = MessageReceiver("nimbus")
 
-    return mysuite
+    filename = os.path.join('/tmp', VIIRS['uid'])
+    touch(filename)
+    string = TwoMetMessage(INPUT_DISPATCH_VIIRS)
+    to_send = msg_rec.receive(string)
+    os.remove(filename)
```

Comment:
The file will be left behind if there's an exception here. Instead I'd do something like

```python
with tempfile.TemporaryDirectory as tmp:
    filename = os.path.join(tmp, VIIRS['uid'])
    touch(filename)
    string = ...
    to_send = ...
```
The context manager will remove also the created temporary file.

Reply:
Oh, and making this a function that takes the UID as argument and returns `to_send` would make it reusable (see below).